src/ui/pages/SignInWithEthereum/SignInWithEthereum.tsx

```diff
@@ -274,18 +276,21 @@ export function SignInWithEthereum() {
             >
               Cancel
             </Button>
-            <SignMessageButton
-              ref={signMsgBtnRef}
-              wallet={wallet}
-              onClick={() => personalSign()}
-              buttonTitle={
-                personalSignMutation.isLoading
-                  ? `Signing In${ellipsis}`
-                  : !siweMessage?.isValid() && isDeviceAccount(wallet)
-                  ? 'Proceed anyway'
-                  : 'Sign In'
-              }
-            />
+            {preferences ? (
+              <SignMessageButton
+                ref={signMsgBtnRef}
+                wallet={wallet}
+                onClick={() => personalSign()}
+                buttonTitle={
+                  personalSignMutation.isLoading
+                    ? `Signing In${ellipsis}`
+                    : !siweMessage?.isValid() && isDeviceAccount(wallet)
+                    ? 'Proceed anyway'
+                    : 'Sign In'
+                }
+                holdToSign={preferences.enableHoldToSignButton}
```

Comment:
doesn't `SignMessageButton` currently read preferences from inside? 🤔  confusing

Reply:
Yes. Because after the last refactor it was extracted from <Content> and now it is able to use useQuery hooks inside it. So I took this opportunity :))